include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,473 @@
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator /(Angle left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary /= operator to scale/assign an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator /=(Angle& left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary / operator to compute the ratio of two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr float operator /(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary % operator to compute modulo of an angle value.
+///
+/// Right hand angle must be greater than zero.
+///
+/// Examples:
+/// \code
+/// sf::degrees(90) % sf::degrees(40)  // 10 degrees
+/// sf::degrees(-90) % sf::degrees(40) // 30 degrees (not -10)
+/// \endcode
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator %(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary %= operator to compute/assign remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator %=(Angle& left, Angle right);
+
+namespace Literals
+{
+
+////////////////////////////////////////////////////////////
+/// \relates sf::Angle
+/// \brief User defined literal for angles in degrees, e.g. 10.5_deg
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates sf::Angle
+/// \brief User defined literal for angles in degrees, e.g. 90_deg
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(unsigned long long angle);
```

Comment:
Wouldn't this technically need to be `unsigned long long int`? 😄

Reply:
No because the standard requires that integral user defined literals use `unsigned long long`. Negative numbers use `operator-` to negate the value as necessary. I believe I wrote test cases for example that situation.

Reply:
Both are the same.

We can even use [`long int unsigned long`](https://www.online-cpp.com/WFLmZv13Tx) 🤦‍♂️

Reply:
Oops sorry I misread what you said, @eXpl0it3r. I can add `int` at the end if that better matches SFML style. I don't believe it technically needs to be `unsigned long long int` though.

Reply:
I just mentioned this, because I've seen that type listed on both these pages:

- https://en.cppreference.com/w/cpp/language/user_literal
- https://docs.microsoft.com/en-us/cpp/cpp/user-defined-literals-cpp?view=msvc-170#user-defined-literal-operator-signatures

In the end it shouldn't really matter, was merely wonder 😅

Reply:
I’ll change that to `unsigned long long int` to matcher better that documentation and match how SFML uses `unsigned int` everywhere instead of `unsigned.`